lib/room.cpp

```diff
@@ -2411,6 +2494,17 @@ Room::Changes Room::Private::addNewMessageEvents(RoomEvents&& events)
 
     QElapsedTimer et;
     et.start();
+
+    //TODO should this be done before dropDuplicateEvents?
```

Comment:
Perhaps not; `dropDuplicateEvents` uses event ids to match events, not the content, and it's better to deduplicate before decrypting.